### PR TITLE
[ADD] implement pos_environment module based on mail_environment

### PR DESCRIPTION
**Pull request by ghost**
pos_environment module for proxy_ip configuration

### CODE REVIEW

File: pos_environment/README.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=========================================
+POS configuration with server_environment
+=========================================
+
+This module allows to configure the hardware proxy (PosBox)
+using the `server_environment` mechanism: you can then have different
```
Comment (yajo) [inline pos_environment/README.rst]: Use double backticks

File: pos_environment/README.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+============
+
+To install this module, you need to have the server_environment module
+installed and properly configured.
+
+Configuration
+=============
+
+With this module installed, the hardware proxy are configured in
+the `server_environment_files` module (which is a module
```
Comment (yajo) [inline pos_environment/README.rst]: double backticks

File: pos_environment/README.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+you should provide, see the documentation of `server_environment` for
```
Comment (yajo) [inline pos_environment/README.rst]: double backticks

File: pos_environment/README.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+more information).
+
+In the configuration file of each environment, you may use the
+section `[hardware_proxy]` to configure the
```
Comment (yajo) [inline pos_environment/README.rst]: double backticks

File: pos_environment/README.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+default values respectively for hardware proxy.
+
+Then for each server, you can define additional values or override the
+default values with a section named `[hardware_proxy.resource_name]`
```
Comment (yajo) [inline pos_environment/README.rst]: double backticks

File: pos_environment/README.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/server-tools/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
+server-tools/issues/new?body=module:%20
+pos_environment%0Aversion:%20
+9.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (yajo) [inline pos_environment/README.rst]: Update links please

File: pos_environment/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (yajo) [inline pos_environment/__openerp__.py]: Update license headers

**Review by yajo — CHANGES_REQUESTED** [review]
Minor details

File: README.md [hunk 8]
```diff
@@ -43,6 +43,7 @@ addon | version | summary
 [mass_editing](mass_editing/) | 9.0.1.0.0 | Mass Editing
 [mass_mailing_enterprise_remove](mass_mailing_enterprise_remove/) | 9.0.1.0.0 | Mass Mailing - Remove Enterprise Features
 [module_prototyper](module_prototyper/) | 9.0.0.1.0 | Prototype your module.
+[pos_environment](pos_environment/) | 9.0.1.0.0 | Configure POS proxy_ip with server_environment_files
```
Comment (simahawk) [inline README.md]: this one is updated automatically AFAIR

**Review by simahawk — CHANGES_REQUESTED** [review]
LGTM except what Jairo already pointed out.

File: pos_environment/views/pos_view.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
```
Comment (leemannd) [inline pos_environment/views/pos_view.xml]: Can be removed if there is no `noupdate`

File: pos_environment/README.rst [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+Images
+------
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Oleksandr Paziuk <oleksandr.paziuk@camptocamp.com>
+* Denis Leemann <denis.leemann@camptocamp.com>
```
Comment (leemannd) [inline pos_environment/README.rst:81]: I just helped you a bit. But thanks

**Review by leemannd — APPROVED** [review]
Only non blocking comment.
Good to me once previous one are fixed.

File: pos_environment/models/env_pos.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    proxy_ip = fields.Char(compute='_compute_server_env',
+                           required=False,
+                           readonly=True)
+
+    @api.depends()
+    def _compute_server_env(self):
```
Comment (simahawk) [inline pos_environment/models/env_pos.py]: broken: you are not setting the value here. You should have `config.proxy_ip = $stuff` at the end

Comment (oleksandrpaziuk) [reply pos_environment/models/env_pos.py]: `config.update(config_vals)` dosen do it?

File: pos_environment/models/pos_config.py [hunk 12]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Copyright 2012-2017 Camptocamp SA
```
Comment (simahawk) [inline pos_environment/models/pos_config.py]: this should be 2017 only :)

**Comment by leemannd** [comment]
Hello @Yajo , it seems like your comments have been answered. Could you update your review?

**Review by leemannd — APPROVED** [review]

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi, 

Sorry to be late on this review, I usually check OCA/pos repository. 

That module is great. Thanks !
I have a question about implementation. For specific settings, you are using the ``name`` of the ``pos.config``. I think it can not works in all cases, specially because generally name belongs spaces or special chars. I don't think that ``[hardware_proxy.My Pos Config]`` will work. Don't you think ?

Two solution I see :
- using ``id`` field instead. (simple, and will works allways)
- keeping ``name`` but cleaned. (replacing spaces, dot, and other special char).

What do you think ?

Otherwise, it will force users to rename of the pos configs that can be a mess. (or maybe not possible sometimes)

regards.

File: pos_environment/models/pos_config.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+from openerp.osv import fields as oldfields
+from openerp import models
+from openerp.addons.server_environment import serv_config
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    _columns = {
```
Comment (legalsylvain) [inline pos_environment/models/pos_config.py:13]: Is there a reason to have an old api syntax for a V9 serie ?

I think that the following line should work.

```

from openerp import models
[...]
proxy_ip = fields.Char(compute=lambda s: s._compute_proxy_ip())

```

Comment (yajo) [reply pos_environment/models/pos_config.py:13]: Indeed, this should be changed as suggested.

File: pos_environment/models/pos_config.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+        if serv_config.has_section(global_section_name):
+            config_vals.update((serv_config.items(global_section_name)))
+
+        if name:
+            custom_section_name = '.'.join((global_section_name, name))
+            if serv_config.has_section(custom_section_name):
+                config_vals.update(serv_config.items(custom_section_name))
+        return config_vals.get(key, '')
+
+    def _compute_proxy_ip(self, cr, uid, ids, fieldnames, args, context=None):
```
Comment (yajo) [inline pos_environment/models/pos_config.py:37]: Use new api.

**Review by yajo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**